### PR TITLE
Refactor code to improve performance and fixed NPE with merge fields contain case sensitive

### PR DESCRIPTION
**Pull request by thangnc**
## CHANGES
- Refactor code to improve performance
- Added configuration param to avoid flooding MailChimp's API
- Fixed bug NPE with merge fields contain case sensitive

### CODE REVIEW

**Comment by thangnc** [comment]
@sakama Can you help me take a look?

File: README.md [hunk 1]
```diff
@@ -51,5 +52,5 @@ out:
 ## Build
 
 ```
-$ rake
+$ ./gradlew package
```
Comment (sakama) [inline README.md]: `$ ./gradlew gem  # -t to watch change of files and rebuild continuously` ?

File: src/main/java/org/embulk/output/mailchimp/helper/MailChimpRetryable.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+                    }
+
+                    @Override
+                    protected boolean isExceptionToRetry(Exception exception)
+                    {
+                        // This check is to make sure if the original exception is retryable, i.e.
+                        // server not found, internal server error...
+                        if (exception instanceof ConfigException || exception instanceof ExecutionException) {
+                            return toRetry((Exception) exception.getCause());
+                        }
```
Comment (sakama) [inline src/main/java/org/embulk/output/mailchimp/helper/MailChimpRetryable.java:98]: Does this block might duplicated with `return (response.getStatus() != 500 && response.getStatus() / 100 != 4) || response.getStatus() == 429;` ?

In my understanding, checking status code?

File: README.md (hunk 1 above)
Comment (thangnc) [reply README.md]: Ah yes, should be `./gradlew gem`. Thanks for catching

File: src/main/java/org/embulk/output/mailchimp/helper/MailChimpRetryable.java (hunk 2 above)
Comment (thangnc) [reply src/main/java/org/embulk/output/mailchimp/helper/MailChimpRetryable.java:98]: There are some errors is not `HttpResponseException`, we will call `super.toRetry`

**Comment by thangnc** [comment]
@sakama PTAL

**Review by sakama — APPROVED** [review]
LGTM

**Comment by thangnc** [comment]
Thank you

**Pull request merged**